addons/custom-zoom/userscript.js

```diff
@@ -0,0 +1,102 @@
+      defaultRect = global.Blockly.getMainWorkspace().zoomControls_.svgGroup_.getBoundingClientRect();
+    }
+
+    try {
+      document.removeEventListener("mousemove", onMouseMove); //Remove the mousemove listener, if it exists
+    } catch {}
+
+    if (e.detail.newUrl.includes("/editor")) document.addEventListener("mousemove", onMouseMove); //If in the editor, add the mousemove listener
+  }
+
+  function hideShow(inArea = false, speed = "default") {
+    //Function to hide/show the zoom controls
+    let speeds = {
+      none: "0",
+      short: "0.25",
+      default: "0.5",
+      long: "1",
+    };
+
+    if (!addon.settings.get("autohide")) return;
+
+    if (!global.Blockly) return;
+    //Get the svg element of the controls
+    let controls = global.Blockly.getMainWorkspace().zoomControls_.svgGroup_;
+
+    controls.style.transition = `${speeds[speed]}s ease-in-out`; //Set the animation speed
+
+    if (!inArea) {
+      //Mouse is not hovering where the controls normally are
+      let val = defaultTranslate;
+      let x = val.substring(val.indexOf("(") + 1, val.indexOf(","));
+      let y = val.substring(val.indexOf(","));
+      x = Number(x) + 80;
+      let translateCode = "translate(" + x.toString() + y;
+
+      controls.attributes.transform.value = translateCode;
+    } else {
+      controls.attributes.transform.value = defaultTranslate || "";
+    }
+  }
+  function onMouseMove(e) {
+    //Function for mousemove listener
+    let val = defaultTranslate;
+    let x = val.substring(val.indexOf("(") + 1, val.indexOf(","));
+    let y = val.substring(val.indexOf(","));
+    x = Number(x) + 80;
+    let translateCode = "translate(" + x.toString() + y;
+    let ctrlTrnsfrm = global.Blockly.getMainWorkspace().zoomControls_.svgGroup_.attributes.transform;
+
+    if (ctrlTrnsfrm.value !== defaultTranslate && ctrlTrnsfrm.value !== translateCode) {
+      let controls = global.Blockly.getMainWorkspace().zoomControls_.svgGroup_;
+      controls.style.transition = `0s ease-in-out`;
+      //Window has been resized and the gui has automatically updated the control position
+      //We now need to update our default values
+      val = ctrlTrnsfrm.value;
+      x = val.substring(val.indexOf("(") + 1, val.indexOf(","));
+      y = val.substring(val.indexOf(","));
+      x = Number(x) + 80;
+      defaultTranslate = controls.attributes.transform.value;
+      controls.attributes.transform.value = defaultTranslate;
+      setTimeout(function () {
+        defaultRect = global.Blockly.getMainWorkspace().zoomControls_.svgGroup_.getBoundingClientRect();
+      }, 0);
+    }
+
+    setZoom({ detail: { newUrl: window.location.href } });
+    hideShow(e.x >= defaultRect.left && e.y >= defaultRect.top, addon.settings.get("speed"));
+  }
+
+  await addon.tab.waitForElement(".blocklyZoom"); //Wait for controls
+  await addon.tab.traps.getBlockly(); //Wait for Blockly
+  global.Blockly = window.Blockly;
```

Comment:
why set global.Blockly variable? Just use "Blockly" in your code instead of "global.Blockly"

Reply:
I tried, it would not work for some reason. It gave me undefined for some part of it, the workspace I think.

Reply:
probably bc you were waiting for Blockly wrong before. Try again and it might work this time.

Reply:
Works now